fdm-app/app/routes/app.addfarm.$b_id_farm.fields.$b_id.tsx

```diff
@@ -0,0 +1,461 @@
+        { name: "description", content: "Welcome to FDM!" },
+    ];
+};
+
+// Form Schema
+const FormSchema = z.object({
+    b_name: z.string({
+        required_error: "Naam van perceel is verplicht",
+    }).min(3, {
+        message: "Naam van perceel moet minimaal 3 karakters bevatten",
+    }),
+    b_lu_catalogue: z.string({
+        required_error: "Hoofdgewas is verplicht",
+    }),
+    // b_soiltype_agr: z.enum(fdmSchema.soilTypes, {
+    //     errorMap: () => ({ message: "Selecteer een grondsoort uit de lijst" })
+    // }),
+    // b_gwl_class: z.enum(fdmSchema.gwlClasses, {
+    //     errorMap: () => ({ message: "Selecteer een grondwatertrap uit de lijst" })
+    // }),
+    b_soiltype_agr: z.string({
+        required_error: "Grondsoort is verplicht",
+    }),
+    b_gwl_class: z.string({
+        required_error: "Grondwatertrap is verplicht",
+    }),
+    a_p_al: z.coerce.number({
+        required_error: "Fosfaat PAL is verplicht",
+    }).gte(1, {
+        message: "Fosfaat PAL moet minimaal 1 zijn",
+    }).lte(250, {
+        message: "Fosfaat PAL mag maximaal 250 zijn",
+    }),
+    a_p_cc: z.coerce.number({
+        required_error: "Fosfaat PAE is verplicht",
+    }).gte(0.1, {
+        message: "Fosfaat PAE moet minimaal 0.1 zijn",
+    }).lte(100, {
+        message: "Fosfaat PAE moet mag maximaal 100 zijn",
+    }),
+    a_som_loi: z.coerce.number({
+        required_error: "Organische stofgehalte is verplicht",
+    }).gte(0.5, {
+        message: "Organische stofgehalte moet minimaal 0.5 zijn",
+    }).lte(75, {
+        message: "Organische stofgehalte mag maximaal 75 zijn",
+    })
+})
+
+/**
+ * Loader function for the field details page.
+ * Fetches field data, soil analysis, cultivation options, and geojson data.
+ * @param request - The incoming request object.
+ * @param params - URL parameters containing `b_id_farm` (farm ID) and `b_id` (field ID).
+ * @returns An object containing field details, cultivation options, and mapbox token.
+ * Throws an error if farm ID, field ID, or field data is missing.
+ */
+export async function loader({
+    request, params
+}: LoaderFunctionArgs) {
+
+    // Get the Id of the farm
+    const b_id_farm = params.b_id_farm
+    if (!b_id_farm) {
+        throw data("Farm ID is required", { status: 400, statusText: "Farm ID is required" });
+    }
+
+    // Get the field id
+    const b_id = params.b_id
+    if (!b_id) {
+        throw data("Field ID is required", { status: 400, statusText: "Field ID is required" });
+    }
+
+    // Get the field data
+    const field = await getField(fdm, b_id)
+    if (!field) {
+        throw data("Field not found", { status: 404, statusText: "Field not found" });
+    }
+
+    // Get the geojson
+    if (!field.b_geometry) {
+        throw data("Field geometry is required", { status: 400, statusText: "Field geometry is required" });
+    }
+    const b_geojson = wkx.Geometry.parse(field.b_geometry).toGeoJSON()
```

Comment:
no checks whether geometry is valid, has correct crs, etc? might be added later.